src/language/dfdl.ts

```diff
@@ -0,0 +1,1328 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode'
+
+export function activate(context: vscode.ExtensionContext) {
+  const elementCompletionProvider =
+    vscode.languages.registerCompletionItemProvider('dfdl', {
+      provideCompletionItems(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        token: vscode.CancellationToken,
+        context: vscode.CompletionContext
+      ) {
+        const wholeLine = document
+          .lineAt(position)
+          .text.substr(0, position.character)
+        if (checkBraceOpen(document, position)) {
+          console.log('in elementCompletionProvider - brace is showing open')
+          return undefined
+        }
+        if (wholeLine.includes('Variable')) {
+          return undefined
+        }
```

Comment:
In this instance, it prevents the auto complete list from containing the defineVariable or setVariable elements if the line where the cursor is already contains a defineVariable or setVariable element.

Reply:
Actually it's more than that. Various completion providers are defined in the code. The completion provider that contains the above code supplies what I considered the base elements. I wanted to limit the items that could be suggested if the line contains defineVariable or setVariable.  This line of code prevents any of these "base" elements from being in the auto complete list for a line containing defineVariable or setVariable. Both defineVariable and setVariable are defined in this completion provider along with numerous other base elements that shouldn't be suggested.

Reply:
The line
```xml
<xs:element name="Variable" dfdl: |
```
would still get autocomplete suggestions when typing `dfdl:`.

As for the second you are correct the autocomplete ignored. Removing this line would cause it to not be ignored I can update that if you prefer. But I do believe we should add in an issue for making for general autocomplete items like `name` will have an autocomplete to `name=""` same with`style` completing to `style=""`.